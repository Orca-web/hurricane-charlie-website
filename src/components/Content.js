import React, { Component } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import Gallery from './Gallery';
import Contact from './Contact';
import Shop from './Shop';

const Content = ({ galleryData }) => {
  return (
    <section className="content-section">
      <Switch>
        <Redirect from="/" to="/gallery" exact />
        <Route
          path="/gallery"
          render={() => <Gallery galleryData={galleryData} />}
        />
        <Route path="/contact" component={Contact} />
        <Route path="/shop" component={Shop} />
      </Switch>
    </section>
  );
}

export default Content;