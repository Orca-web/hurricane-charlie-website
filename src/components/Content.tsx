import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import Gallery from './Gallery';
import Contact from './Contact';
import Shop from './Shop';
import ErrorPage from './ErrorPage';

const Content = (): JSX.Element => (
    <section className="content-section">
        <Switch>
            <Redirect from="/" to="/gallery" exact />
            <Route path="/gallery" component={Gallery} />
            <Route path="/contact" component={Contact} />
            <Route path="/shop" component={Shop} />
            <Route component={ErrorPage} />
        </Switch>
    </section>
);

export default Content;