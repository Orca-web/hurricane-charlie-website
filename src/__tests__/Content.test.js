import React from 'react';
import { shallow } from 'enzyme';
import Content from '../components/Content';

const app = shallow(<Content />);

describe('Content Section', () => {
  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });
});