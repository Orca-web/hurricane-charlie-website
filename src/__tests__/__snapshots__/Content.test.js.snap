// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Section renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Redirect
          exact={true}
          from="/"
          push={false}
          to="/gallery"
        />
        <Route
          path="/gallery"
          render={[Function]}
        />
        <Route
          component={[Function]}
          path="/contact"
        />
        <Route
          component={[Function]}
          path="/shop"
        />
        <Route
          component={[Function]}
        />
      </Switch>,
      "className": "content-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Redirect
            exact={true}
            from="/"
            push={false}
            to="/gallery"
          />,
          <Route
            path="/gallery"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            path="/contact"
          />,
          <Route
            component={[Function]}
            path="/shop"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "from": "/",
            "push": false,
            "to": "/gallery",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/gallery",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/contact",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/shop",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Redirect
            exact={true}
            from="/"
            push={false}
            to="/gallery"
          />
          <Route
            path="/gallery"
            render={[Function]}
          />
          <Route
            component={[Function]}
            path="/contact"
          />
          <Route
            component={[Function]}
            path="/shop"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        "className": "content-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Redirect
              exact={true}
              from="/"
              push={false}
              to="/gallery"
            />,
            <Route
              path="/gallery"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              path="/contact"
            />,
            <Route
              component={[Function]}
              path="/shop"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "from": "/",
              "push": false,
              "to": "/gallery",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/gallery",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/shop",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
